docs/src/main/tut/codecs/adt.md

```diff
@@ -21,25 +21,27 @@ case class Qux(values: List[String]) extends Event
 And the encoder / decoder instances:
 
 ```scala mdoc:silent
+import cats.data.NonEmptyList
 import cats.syntax.functor._
+import cats.syntax.reducible._
 import io.circe.{ Decoder, Encoder }, io.circe.generic.auto._
 import io.circe.syntax._
 
 object GenericDerivation {
   implicit val encodeEvent: Encoder[Event] = Encoder.instance {
-    case foo @ Foo(_) => foo.asJson
-    case bar @ Bar(_) => bar.asJson
-    case baz @ Baz(_) => baz.asJson
-    case qux @ Qux(_) => qux.asJson
+    case foo: Foo => foo.asJson
+    case bar: Bar => bar.asJson
+    case baz: Baz => baz.asJson
+    case qux: Qux => qux.asJson
   }
 
   implicit val decodeEvent: Decoder[Event] =
-    List[Decoder[Event]](
+    NonEmptyList.of[Decoder[Event]](
       Decoder[Foo].widen,
       Decoder[Bar].widen,
       Decoder[Baz].widen,
       Decoder[Qux].widen
-    ).reduceLeft(_ or _)
+    ).reduceK
 }
```

Comment:
I don't think for the target audience of our documentation, that the use of `reduceK` over `reduceLeft(_ or _)` actually makes it any more clear on how we might compose decoders.

Reply:
no problems, I'll remove it. I just believed that the docs should provide with, I'd say, the most "correct" code (and here passing the function explicitly is kind of redundant)

Reply:
moreover, this code (with the `reduceLeft` on the list) can be considered as erroneous by the scapegoat's `UnsafeTraversableMethods` inspection

Reply:
removed

Reply:
Thanks for updating it.  I didn't think about the linter scapegoat, and the `UnsafeTraversableMethods`.  I think switching to a `NonEmptyList` would fix that specific issue, and showing another example with `reduceK` could be fine.  I just want to make sure our average audience understands at a glance what is going on.